.pre-commit-config.yaml

```diff
@@ -1,37 +1,40 @@
+
 repos:
-  - repo: local
+-   repo: local
     hooks:
-      - id: isort
-        name: isort
-        stages: [commit]
-        types: [python]
-        entry: poetry run isort .
-        language: system
-        pass_filenames: false
-        always_run: true
-      - id: black
-        name: black
-        stages: [commit]
-        types: [python]
-        entry: poetry run black .
-        language: system
-        pass_filenames: false
-        always_run: true
-      - id: ruff
-        name: ruff
-        stages: [commit]
-        types: [python]
-        entry: poetry run ruff .
-        language: system
-        pass_filenames: false
-        always_run: true
-        fail_fast: true
-      - id: pytest
-        name: pytest
-        stages: [commit]
-        types: [python]
-        entry: poetry run pytest
-        language: system
-        pass_filenames: false
-        always_run: true
-        fail_fast: true
+    - id: rye-lint
+      name: rye-lint
+      description: "Lint Python via 'rye lint'"
+      entry: rye lint --fix
+      language: system
+      types_or: [python, pyi]
+      args: []
+      require_serial: true
+      additional_dependencies: []
+      minimum_pre_commit_version: "2.9.2"
+    - id: rye-format
+      name: rye-format
+      description: "Format Python via 'rye fmt'"
+      entry: rye fmt
+      language: system
+      types_or: [python, pyi]
+      args: []
+      require_serial: true
+      additional_dependencies: []
+      minimum_pre_commit_version: "2.9.2"
+    - id: rye-test
```

Comment:
I assume this is because otherwise you'd slow down velocity with tests running each time you commit, while you want commits to be frequent and no-brainers?